From: Chad Dupuis <cdupuis@redhat.com>
Date: Tue, 20 Jul 2010 15:31:29 -0400
Subject: [net] qlcnic: use DEFINE_PCI_DEVICE_TABLE
Message-id: <20100720153035.2978.25131.sendpatchset@localhost.localdomain>
Patchwork-id: 26961
O-Subject: [RHEL 5.6 PATCH 2/16] qlcnic: use DEFINE_PCI_DEVICE_TABLE
Bugzilla: 562723
RH-Acked-by: David S. Miller <davem@redhat.com>

Bugzilla
========

562723

Upstream Status
===============

net-2.6 commit id 6a902881cc6993f136906559ca04b6a4c9e260fc

Description
===========

>From 1b5d2ce5783155036d17843495f2c0932708971f Mon Sep 17 00:00:00 2001
From: Amit Kumar Salecha <amit.salecha@qlogic.com>
Date: Mon, 1 Feb 2010 05:24:54 +0000
Subject: [PATCH] qlcnic: use DEFINE_PCI_DEVICE_TABLE

Use DEFINE_PCI_DEVICE_TABLE() so we get place PCI ids table into
correct section in every case.

Signed-off-by: Amit Kumar Salecha <amit.salecha@qlogic.com>
Signed-off-by: David S. Miller <davem@davemloft.net>
Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/net/qlcnic/qlcnic_main.c b/drivers/net/qlcnic/qlcnic_main.c
index af86196..7cbf74b 100644
--- a/drivers/net/qlcnic/qlcnic_main.c
+++ b/drivers/net/qlcnic/qlcnic_main.c
@@ -103,7 +103,7 @@ static void qlcnic_config_indev_addr(struct net_device *dev, unsigned long);
 
 #define PCI_DEVICE_ID_QLOGIC_QLE824X  0x8020
 
-static const struct pci_device_id qlcnic_pci_tbl[] __devinitdata = {
+static DEFINE_PCI_DEVICE_TABLE(qlcnic_pci_tbl) = {
 	ENTRY(PCI_DEVICE_ID_QLOGIC_QLE824X),
 	{0,}
 };
